src/Smalot/PdfParser/Font.php

```diff
@@ -414,18 +414,25 @@ protected function getFontSpaceLimit(): int
     }
 
     /**
-     * Decode text by commands array.
+     * Decode text by commands array. decodeText now inserts the proper
+     * whitespace by examining the scaling of the current text matrix.
+     * Supply the identity matrix '1 0 0 1' by default.
      */
-    public function decodeText(array $commands): string
-    {
+    public function decodeText(
+        array $commands,
+        array $textMatrix = ['a' => 1, 'b' => 0, 'i' => 0, 'j' => 1]
+    ): string {
         $word_position = 0;
         $words = [];
+
         $font_space = $this->getFontSpaceLimit();
+        $font_space = $font_space * $textMatrix['a'] + $font_space * $textMatrix['b'];
```

Comment:
Can we be sure that key `a` and `b` are always set? Please add the explanation above the line.